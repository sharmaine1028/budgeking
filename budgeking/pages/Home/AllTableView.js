import React from "react";
import { StyleSheet, ScrollView, View, Text } from "react-native";
import { Header, Title } from "../../config/reusableText";
import colours from "../../config/colours";
import Icon from "react-native-vector-icons/AntDesign";
import { auth, db } from "../../config/firebase";
import { BlackButton } from "../../config/reusableButton";
import { GreyLine } from "../../config/reusablePart";
import { dateFormat, categoryFormat, timeFormat } from "../Home/HomePage";
import { renderNoRecords } from "../Home/HomePage";

class AllTableView extends React.Component {
  constructor() {
    super();
    this.ExpenseRef = db
      .collection("users")
      .doc(auth.currentUser.uid)
      .collection("expense");
    this.IncomeRef = db
      .collection("users")
      .doc(auth.currentUser.uid)
      .collection("income");
    this.state = {
      expenseArr: [],
      incomeArr: [],
      budget: "Expense",
    };
  }

  componentDidMount() {
    this.unsubscribeExpenseRef = this.ExpenseRef.onSnapshot(
      this.getCollectionExpense
    );
    this.unsubscribeIncomeRef = this.IncomeRef.onSnapshot(
      this.getCollectionIncome
    );
  }

  componentWillUnmount() {
    this.unsubscribeExpenseRef();
    this.unsubscribeIncomeRef();
  }

  getCollectionExpense = (querySnapshot) => {
    const expenseArrPush = [];
    querySnapshot.forEach((res) => {
      const { notes, value, category, date, time, photoURL, address } =
        res.data();
      expenseArrPush.push({
        key: res.id,
        value,
        notes,
        category,
        date,
        time,
        photoURL,
        address,
      });
    });
    this.setState({
      expenseArr: expenseArrPush,
    });
  };

  getCollectionIncome = (querySnapshot) => {
    const incomeArrPush = [];
    querySnapshot.forEach((res) => {
      const { category, date, notes, value, time, photoURL, address } =
        res.data();
      incomeArrPush.push({
        key: res.id,
        value,
        category,
        notes,
        date,
        time,
        photoURL,
        address,
      });
    });
    this.setState({
      incomeArr: incomeArrPush,
    });
  };

  sortedArr(arr) {
    let sortedArr = arr.sort((a, b) => {
      if (dateFormat(a.date.seconds) == dateFormat(b.date.seconds)) {
        return b.time.seconds - a.time.seconds;
      } else {
        return b.date.seconds - a.date.seconds;
      }
    });
    return sortedArr;
  }

  generatePhotoAttached(doc) {
    console.log("photo", doc.photoURL);
    return (
      <View>
        <Text>{doc.photoURL}</Text>
        <Image
          source={{ uri: doc.photoURL }}
          style={{ width: 200, height: 200 }}
        />
        <TouchableOpacity onPress={() => this.toggleModal()}>
          <Image
            style={{
              width: 50,
              height: 50,
            }}
            source={{ uri: doc.photoURL }}
          />
        </TouchableOpacity>

        <View style={styles.modalView}>
          <Modal visible={this.state.showModal} animated>
            <View style={styles.modalView}>
              <View style={styles.modal}>
                <TouchableOpacity onPress={() => this.toggleModal()}>
                  <MaterialCommunityIcons
                    name="close-circle"
                    size={30}
                    color="black"
                    style={styles.logo}
                  />
                </TouchableOpacity>

                <Image
                  style={{
                    resizeMode: "contain",
                    width: 350,
                    height: 350,
                  }}
                  source={{ uri: doc.photoURL }}
                />
              </View>
            </View>
          </Modal>
        </View>
      </View>
    );
  }

  generatePhotoAttached(doc) {
    console.log("photo", doc.photoURL);
    return (
      <View>
        <Text>{doc.photoURL}</Text>
        <Image
          source={{ uri: doc.photoURL }}
          style={{ width: 200, height: 200 }}
        />
        <TouchableOpacity onPress={() => this.toggleModal()}>
          <Image
            style={{
              width: 50,
              height: 50,
            }}
            source={{ uri: doc.photoURL }}
          />
        </TouchableOpacity>

        <View style={styles.modalView}>
          <Modal visible={this.state.showModal} animated>
            <View style={styles.modalView}>
              <View style={styles.modal}>
                <TouchableOpacity onPress={() => this.toggleModal()}>
                  <MaterialCommunityIcons
                    name="close-circle"
                    size={30}
                    color="black"
                    style={styles.logo}
                  />
                </TouchableOpacity>

                <Image
                  style={{
                    resizeMode: "contain",
                    width: 350,
                    height: 350,
                  }}
                  source={{ uri: doc.photoURL }}
                />
              </View>
            </View>
          </Modal>
        </View>
      </View>
    );
  }

  generateExpensesIncome = (doc) => {
    return (
      <View key={doc.key} style={styles.row}>
        <View style={styles.dateRow}>
          <Icon.Button
            name={this.state.budget == "Expense" ? "arrowright" : "arrowleft"}
            size={20}
            color={
              this.state.budget == "Expense" ? colours.tomato : colours.green
            }
            backgroundColor={"transparent"}
            iconStyle={{ marginLeft: 5, marginRight: 0 }}
          />
          <Header
            text={`${dateFormat(doc.date.seconds)}`}
            style={{ fontWeight: "bold", marginTop: 12 }}
          />
        </View>
        <GreyLine />

        <View style={styles.categoryRow}>
          <Header
            text={categoryFormat(doc.category)}
            style={styles.expenseCategory}
          />

          <Text style={styles.valueText}>{`$${doc.value}`}</Text>
        </View>

        <View style={styles.notesRow}>
          <Text style={styles.noteText}>{doc.notes}</Text>
          <Text style={styles.timeText}>{timeFormat(doc.time.seconds)}</Text>
        </View>

        <GreyLine />
        <View style={styles.dateRow}>
          <Image
            source={require("../../assets/addphoto.png")}
            style={styles.image}
          />
          {doc.photoURL != "" ? (
            this.generatePhotoAttached(doc)
          ) : (
            <Text style={{ marginTop: 8, fontWeight: "200", fontSize: 14 }}>
              No photo attached
            </Text>
          )}
        </View>
        <View style={styles.dateRow}>
          <Image
            source={require("../../assets/location.png")}
            style={styles.image}
          />
          {doc.address != "" ? (
            <Text style={{ marginTop: 8, fontSize: 14, color: colours.black }}>
              {doc.address}
            </Text>
          ) : (
            <Text style={{ marginTop: 8, fontWeight: "200", fontSize: 14 }}>
              No location
            </Text>
          )}
        </View>
      </View>
    );
  };

  whatBudget() {
    if (this.state.budget == "Expense" && this.state.expenseArr.length !== 0) {
      return this.sortedArr(this.state.expenseArr).map((doc) =>
        this.generateExpensesIncome(doc)
      );
    } else if (
      this.state.budget == "Income" &&
      this.state.incomeArr.length !== 0
    ) {
      return this.sortedArr(this.state.incomeArr).map((doc) =>
        this.generateExpensesIncome(doc)
      );
    } else {
      return renderNoRecords();
    }
  }

  budgetButtons = () => {
    return (
      <View style={styles.tabContainer}>
        <BlackButton
          text={"Expenses"}
          style={styles.expenseButton}
          textStyle={{ color: colours.black }}
          onPress={() => {
            this.setState({ budget: "Expense" });
          }}
        />
        <BlackButton
          text={"Income"}
          style={styles.incomeButton}
          textStyle={{ color: colours.black }}
          onPress={() => this.setState({ budget: "Income" })}
        />
      </View>
    );
  };

  render() {
    return (
      <View style={styles.container}>
        <ScrollView>
          <View style={styles.transactionTitle}>
            <Icon.Button
              name="profile"
              size={30}
              color={colours.black}
              backgroundColor={"transparent"}
              iconStyle={{ alignSelf: "center" }}
            />
            <Title
              style={styles.transactionText}
              text={"Transaction History"}
            />
          </View>
          {this.budgetButtons()}
          {this.whatBudget()}
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex",
    marginHorizontal: 20,
    backgroundColor: colours.white,
  },
  transactionTitle: {
    marginTop: 20,
    flexDirection: "row",
    // borderWidth: 4,
    justifyContent: "center",
  },
  transactionText: {
    marginTop: 15,
    textAlign: "center",
  },
  tabContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  expenseButton: {
    backgroundColor: "#ffb8ac",
    flexGrow: 0.5,
  },
  incomeButton: {
    backgroundColor: colours.green,
    flexGrow: 0.5,
  },
  row: {
    backgroundColor: colours.beige,
    borderWidth: 2,
    borderColor: colours.lightBrown,
    marginTop: 15,
    borderRadius: 10,
  },
  dateRow: {
    flexDirection: "row",
    marginBottom: 0,
  },
  categoryRow: {
    flexDirection: "row",
    alignContent: "flex-start",
    justifyContent: "space-between",
  },
  notesRow: {
    flexDirection: "row",
    alignContent: "flex-end",
    justifyContent: "space-between",
  },
  expenseCategory: { marginLeft: 10, marginTop: 1 },
  valueText: {
    marginRight: 10,
    marginTop: 1,
    color: colours.black,
    textAlign: "right",
  },
  noteText: {
    fontWeight: "200",
    marginLeft: 10,
    marginBottom: 10,
  },
  timeText: {
    fontWeight: "200",
    marginRight: 10,
    marginBottom: 10,
  },
});

export default AllTableView;
