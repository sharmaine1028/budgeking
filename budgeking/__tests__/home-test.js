import { dateFormat, categoryFormat, timeFormat } from "../pages/Home/HomePage";
import { renderNoRecords, generate3ExpensesLR } from "../pages/Home/HomePage";
import renderer from "react-test-renderer";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";

// dateFormat
test("date formats correctly", () => {
  const dateMilliseconds = new Date("Dec 12, 2022 23:15:30").getTime();
  const convertToSeconds = dateMilliseconds / 1000;
  const dateMilliseconds1 = new Date("May 9, 2001 23:15:30").getTime();
  const convertToSeconds1 = dateMilliseconds1 / 1000;
  expect(dateFormat(convertToSeconds)).toBe("12 Dec 2022"); // 12 Dec 2022
  expect(dateFormat(convertToSeconds1)).toBe("9 May 2001"); // 9 May 2001
});

// categoryFormat
test("category formats correctly", () => {
  const foodDrink = "food and drinks";
  const edu = "education";
  expect(categoryFormat(foodDrink)).toBe("Food & Drinks");
  expect(categoryFormat(edu)).toBe("Education");
});

// timeFormat
test("time formats correctly", () => {
  const dateMillisecondsPM = new Date("Dec 12, 2022 23:59:30").getTime();
  const convertToSecondsPM = dateMillisecondsPM / 1000;
  const dateMillisecondsAM = new Date("Dec 12, 2022 9:09:30").getTime();
  const convertToSecondsAM = dateMillisecondsAM / 1000;
  const dateMilliseconds0 = new Date("Dec 12, 2022 0:00:00").getTime();
  const convertToSeconds0 = dateMilliseconds0 / 1000;
  const dateMillisecondsNoon = new Date("Dec 12, 2022 12:00:00").getTime();
  const convertToSecondsNoon = dateMillisecondsNoon / 1000;
  expect(timeFormat(convertToSecondsPM)).toBe("11:59 PM");
  expect(timeFormat(convertToSecondsAM)).toBe("9:09 AM");
  expect(timeFormat(convertToSeconds0)).toBe("12:00 AM");
  expect(timeFormat(convertToSecondsNoon)).toBe("12:00 PM");
});

// renderNoRecords
test("Table view renders no records yet if empty database", () => {
  const expenses = [];
  const tree = renderer.create(
    expenses.length !== 0
      ? expenses.map((doc) => generate3ExpensesLR(doc))
      : renderNoRecords()
  );
  expect(tree).toMatchSnapshot();
});

// generate3ExpensesLR
test("expenses renders correctly with data", () => {
  const expenses = [
    (item = {
      category: "food and drinks",
      date: {
        nanoseconds: 591000000,
        seconds: 1658382053,
      },
      key: "VBjtiejjNZO2yLUV2nGj",
      notes: "",
      time: {
        nanoseconds: 591000000,
        seconds: 1658382053,
      },
      value: 10.5,
    }),
    (item = {
      category: "health",
      date: {
        nanoseconds: 868000000,
        seconds: 1658122274,
      },
      key: "lwX2I0eq9oYaWL01nbW3",
      notes: "Annual detailed checkup",
      time: {
        nanoseconds: 402000000,
        seconds: 1658179874,
      },
      value: 60,
    }),
    (item = {
      category: "health",
      date: {
        nanoseconds: 0,
        seconds: 1624202293,
      },
      key: "S10W60mbFClRQ2DZifxP",
      notes: "Uniqlo ",
      time: {
        nanoseconds: 0,
        seconds: 1655695093,
      },
      value: 12.78,
    }),
  ];
  const ex = () => {
    expenses.map((doc) => generate3ExpensesLR(doc));
  };
  const Stack = createStackNavigator();
  const MockNavigator = (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="MockedScreen" component={ex} />
      </Stack.Navigator>
    </NavigationContainer>
  );

  jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: jest.fn(),
        dispatch: jest.fn(),
      }),
    };
  });
  const tree = renderer
    .create(
      expenses.length !== 0
        ? expenses.map((doc) => generate3ExpensesLR(doc))
        : renderNoRecords()
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

expenseArrayTimeConverted = [
  {
    category: "food and drinks",
    date: {
      nanoseconds: 591000000,
      seconds: 1658382053,
    },
    key: "VBjtiejjNZO2yLUV2nGj",
    notes: "",
    time: {
      nanoseconds: 591000000,
      seconds: 1658382053,
    },
    value: 10,
  },
  {
    category: "transportation",
    date: {
      nanoseconds: 271000000,
      seconds: 1658388773,
    },
    key: "ZWcspIIBTGGdrF7SG9wf",
    notes: "",
    time: {
      nanoseconds: 0,
      seconds: 1658376053,
    },
    value: 15.55,
  },
  {
    category: "food and drinks",
    date: {
      nanoseconds: 328000000,
      seconds: 1658331027,
    },
    key: "2hxBUn7RvQwg99pSq8Aj",
    notes: "",
    time: {
      nanoseconds: 328000000,
      seconds: 1658331027,
    },
    value: 12,
  },
  {
    category: "education",
    date: {
      nanoseconds: 0,
      seconds: 1658229345,
    },
    key: "fCyTvlqG6ymIrvqjNRz1",
    notes: "",
    time: {
      nanoseconds: 174000000,
      seconds: 1658315745,
    },
    value: 20,
  },
  {
    category: "health",
    date: {
      nanoseconds: 868000000,
      seconds: 1658122274,
    },
    key: "lwX2I0eq9oYaWL01nbW3",
    notes: "Annual detailed checkup",
    time: {
      nanoseconds: 402000000,
      seconds: 1658179874,
    },
    value: 60,
  },
  {
    category: "food and drinks",
    date: {
      nanoseconds: 0,
      seconds: 1658036981,
    },
    key: "eka9wiAzI1Dvzf3BLo6k",
    notes: "",
    time: {
      nanoseconds: 0,
      seconds: 1658191781,
    },
    value: 1.79,
  },
  {
    category: "others",
    date: {
      nanoseconds: 195000000,
      seconds: 1657744387,
    },
    key: "Ruvwoyeur91Ww87yeb7L",
    notes: "",
    time: {
      nanoseconds: 195000000,
      seconds: 1657744387,
    },
    value: 0.45,
  },
  {
    category: "food and drinks",
    date: {
      nanoseconds: 825000000,
      seconds: 1657743891,
    },
    key: "l3Q90QTVFMWpExKGL9LE",
    notes: "",
    time: {
      nanoseconds: 825000000,
      seconds: 1657743891,
    },
    value: 0.43,
  },
  {
    category: "shopping",
    date: {
      nanoseconds: 673000000,
      seconds: 1656912459,
    },
    key: "RFJNtPN7zq8RwWQL4RcJ",
    notes: "",
    time: {
      nanoseconds: 619000000,
      seconds: 1658122059,
    },
    value: 12.22,
  },
  {
    category: "shopping",
    date: {
      nanoseconds: 576000000,
      seconds: 1656080421,
    },
    key: "WjLx5SQnTHjKxxKfn4NW",
    notes: "Uniqlo",
    time: {
      nanoseconds: 278000000,
      seconds: 1656080421,
    },
    value: 12,
  },
  {
    category: "health",
    date: {
      nanoseconds: 121000000,
      seconds: 1655968071,
    },
    key: "UXhKu3Nk5nOi6PdeHMt2",
    notes: "uhc",
    time: {
      nanoseconds: 0,
      seconds: 1655924871,
    },
    value: 12,
  },
  {
    category: "transportation",
    date: {
      nanoseconds: 68000000,
      seconds: 1655699341,
    },
    key: "8fvXByUNztNXUYtf1kBT",
    notes: "Top up card",
    time: {
      nanoseconds: 68000000,
      seconds: 1655699341,
    },
    value: 20,
  },
  {
    category: "others",
    date: {
      nanoseconds: 668000000,
      seconds: 1652885839,
    },
    key: "yYbJvvhXz7zkQbxVZB4B",
    notes: "",
    time: {
      nanoseconds: 654000000,
      seconds: 1652885839,
    },
    value: 20.5,
  },
  {
    category: "housing",
    date: {
      nanoseconds: 0,
      seconds: 1624641464,
    },
    key: "4eJR7VhR4UjotkVWeYyo",
    notes: "",
    time: {
      nanoseconds: 657000000,
      seconds: 1656177464,
    },
    value: 100,
  },
  {
    category: "health",
    date: {
      nanoseconds: 0,
      seconds: 1624202293,
    },
    key: "S10W60mbFClRQ2DZifxP",
    notes: "Uniqlo ",
    time: {
      nanoseconds: 0,
      seconds: 1655695093,
    },
    value: 12.78,
  },
];
