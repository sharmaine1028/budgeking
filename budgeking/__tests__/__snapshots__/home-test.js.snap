// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing home components expenses renders correctly with data 1`] = `
Array [
  <View
    style={
      Object {
        "borderRadius": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 15,
              "marginVertical": 5,
            },
            Object {
              "fontWeight": "bold",
              "marginTop": 12,
            },
          ]
        }
      >
        21 Jul 2022
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontWeight": "200",
                "marginBottom": 10,
                "marginRight": 10,
              },
              Object {
                "alignSelf": "flex-end",
                "marginTop": 12,
              },
            ]
          }
        >
          1:40 PM
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignContent": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 15,
              "marginVertical": 5,
            },
            Object {
              "marginLeft": 10,
              "marginTop": 1,
            },
          ]
        }
      >
        Food & Drinks
      </Text>
      <Text
        style={
          Object {
            "color": "#413E3E",
            "marginRight": 10,
            "marginTop": 1,
            "textAlign": "right",
          }
        }
      >
        $10.5
      </Text>
    </View>
    <View
      style={
        Object {
          "alignContent": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "200",
            "marginBottom": 10,
            "marginLeft": 10,
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E0E0E0",
          "borderBottomWidth": 2,
          "marginVertical": 3,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "borderRadius": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 15,
              "marginVertical": 5,
            },
            Object {
              "fontWeight": "bold",
              "marginTop": 12,
            },
          ]
        }
      >
        18 Jul 2022
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontWeight": "200",
                "marginBottom": 10,
                "marginRight": 10,
              },
              Object {
                "alignSelf": "flex-end",
                "marginTop": 12,
              },
            ]
          }
        >
          5:31 AM
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignContent": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 15,
              "marginVertical": 5,
            },
            Object {
              "marginLeft": 10,
              "marginTop": 1,
            },
          ]
        }
      >
        Health
      </Text>
      <Text
        style={
          Object {
            "color": "#413E3E",
            "marginRight": 10,
            "marginTop": 1,
            "textAlign": "right",
          }
        }
      >
        $60
      </Text>
    </View>
    <View
      style={
        Object {
          "alignContent": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "200",
            "marginBottom": 10,
            "marginLeft": 10,
          }
        }
      >
        Annual detailed checkup
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E0E0E0",
          "borderBottomWidth": 2,
          "marginVertical": 3,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "borderRadius": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 15,
              "marginVertical": 5,
            },
            Object {
              "fontWeight": "bold",
              "marginTop": 12,
            },
          ]
        }
      >
        20 Jun 2021
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontWeight": "200",
                "marginBottom": 10,
                "marginRight": 10,
              },
              Object {
                "alignSelf": "flex-end",
                "marginTop": 12,
              },
            ]
          }
        >
          11:18 AM
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignContent": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 15,
              "marginVertical": 5,
            },
            Object {
              "marginLeft": 10,
              "marginTop": 1,
            },
          ]
        }
      >
        Health
      </Text>
      <Text
        style={
          Object {
            "color": "#413E3E",
            "marginRight": 10,
            "marginTop": 1,
            "textAlign": "right",
          }
        }
      >
        $12.78
      </Text>
    </View>
    <View
      style={
        Object {
          "alignContent": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "200",
            "marginBottom": 10,
            "marginLeft": 10,
          }
        }
      >
        Uniqlo 
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E0E0E0",
          "borderBottomWidth": 2,
          "marginVertical": 3,
        }
      }
    />
  </View>,
]
`;

exports[`testing home components pie chart renders legend correctly 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 12,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#177AD5",
          "borderRadius": 4,
          "height": 18,
          "marginRight": 10,
          "width": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#444444",
          "fontSize": 16,
        }
      }
    >
      Food & Drinks
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 12,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#79D2DE",
          "borderRadius": 4,
          "height": 18,
          "marginRight": 10,
          "width": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#444444",
          "fontSize": 16,
        }
      }
    >
      Transportation
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 12,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F7D8B5",
          "borderRadius": 4,
          "height": 18,
          "marginRight": 10,
          "width": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#444444",
          "fontSize": 16,
        }
      }
    >
      Housing
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 12,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#8F80E4",
          "borderRadius": 4,
          "height": 18,
          "marginRight": 10,
          "width": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#444444",
          "fontSize": 16,
        }
      }
    >
      Shopping
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 12,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FB8875",
          "borderRadius": 4,
          "height": 18,
          "marginRight": 10,
          "width": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#444444",
          "fontSize": 16,
        }
      }
    >
      Health
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 12,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FDE74C",
          "borderRadius": 4,
          "height": 18,
          "marginRight": 10,
          "width": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#444444",
          "fontSize": 16,
        }
      }
    >
      Education
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 12,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E8E0CE",
          "borderRadius": 4,
          "height": 18,
          "marginRight": 10,
          "width": 18,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#444444",
          "fontSize": 16,
        }
      }
    >
      Others
    </Text>
  </View>,
]
`;

exports[`testing home components table view renders no records yet if empty database 1`] = `
<Text
  style={
    Object {
      "alignSelf": "center",
      "marginTop": 20,
    }
  }
>
  No Records Yet
</Text>
`;
